### PR TITLE
Using DBCA option -autoGeneratePasswords while creating the database

### PR DESCRIPTION
**Pull request by abhisbyk**
For 21c, DBCA option `-autoGeneratePasswords` used for generating passwords automatically while creating the database. The user has to change this password **mandatorily** using `setpassword.sh` script after the database is created and the corresponding container is healthy. This new password will be used afterwards by the user to connect with the database.

### CODE REVIEW

**Comment by abhisbyk** [comment]
@gvenzl The PR has been corrected. DBCA option `-autoGeneratePasswords` is used only when the user does not provide the password using ORACLE_PWD env var.
If user provide the password using ORACLE_PWD, then this password will be used for the database creation. Please review again.

Raised this PR for bug no. 33355394.

**Review by yunus-qureshi — DISMISSED** [review]
Seems fine, thanks

**Review by Mahi0911 — DISMISSED** [review]

**Comment by Mahi0911** [comment]
Looks Fine, Thanks

File: OracleDatabase/SingleInstance/dockerfiles/21.3.0/createDB.sh [hunk 1]
```diff
@@ -71,23 +71,27 @@ if [[ "${INIT_SGA_SIZE}" != "" && "${INIT_PGA_SIZE}" == "" ]] || [[ "${INIT_SGA_
...
 else
   if [[ "${CLONE_DB}" == "true" ]] || [[ "${STANDBY_DB}" == "true" ]]; then
+    # Validation: Checking if ORACLE_PWD is provided or not
+    if [[ -z "$ORACLE_PWD" ]]; then
+      echo "ERROR: Please provide sys password of the primary database as ORACLE_PWD env variable. Exiting..."
+      exit 1
+    fi
+
     # Creating temporary response file containing sysPassword for clone/standby cases
     echo "sysPassword=${ORACLE_PWD}" > $ORACLE_BASE/dbca.rsp
```
Comment (yunus-qureshi) [inline OracleDatabase/SingleInstance/dockerfiles/21.3.0/createDB.sh:87]: umask 600
echo

**Comment by abhisbyk** [comment]
Incorporated fixes regarding security bug 33355394. 

- Used `umask` for masking the permissions before new file creation. 
- Providing the password using here document.

Also, as previously mentioned, dbca option -autoGeneratePasswords is used only when user does not provide the password using ORACLE_PWD.
Please review.

**Review by yunus-qureshi — APPROVED** [review]

**Review by gvenzl — DISMISSED** [review]

**Review by yunus-qureshi — APPROVED** [review]

**Review by gvenzl — APPROVED** [review]

**Pull request merged**